pkg/engine/path_selection/path_selection.go

```diff
@@ -83,6 +84,24 @@ func BuildPathSelectionGraph(
 		if !PathSatisfiesClassification(kb, resourcePath, classification) {
 			continue
 		}
+		// Check to see if the whole path is valid before adding phantoms and edges.
+		// It's a miniscule performance benefit, and is mostly done for clarity in the debug graph output.
+		validPath := true
+		for i, res := range path {
+			if i == 0 {
+				continue
+			}
+			edgeTemplate := kb.GetEdgeTemplate(path[i-1].Id(), res.Id())
+			if edgeTemplate == nil || edgeTemplate.DirectEdgeOnly {
+				validPath = false
+				break
+			}
+		}
+		if !validPath {
+			continue
+		}
+
+		satisfied_paths++
```

Comment:
dont we already do this when building the path selection graph? Isnt this redundant

Reply:
Ah, yeah, I meant to move it instead of duplicate it. The main impact of this is that it removed resources used in paths that never connected to the target. Per the comment, it's mostly to support the `kb` tool to prevent extra irrelevant resources.